types/sinon/sinon-tests.ts

```diff
@@ -754,3 +754,61 @@ async function testPromises() {
         reject('some error');
     });
 }
+
+async function testTypedFake() {
+    const fake1: sinon.SinonSpy<[number, string], boolean> = sinon.fake((arg1, arg2) => {
```

Comment:
can you make sure there's an equivalent test for each of these typed tests which use the default type parameters?

e.g. calling fake with a strongly typed function and asserting the inferred types are correct (args and return value)